### ISSUE TITLE
Node architecture label shouldn't be hard-coded to equal OS

### ISSUE THREAD

**Issue by nightkr** [deislabs/krustlet#684]
https://github.com/krustlet/krustlet/blob/a6efe8ad0cbf81939b8a80b98763318e91314512/crates/kubelet/src/node/mod.rs#L447-L454

In general, this seems to be set up to be `linux` in both cases, which means that you can't schedule pods to specific architectures in multiarch clusters.

For a regular Kubelet this would typically be set up so that `os: linux, arch: amd64`.

**Comment by teozkr** [comment]
Yeah, for Krustlet-the-app I'd expect it to be hard-coded to something like `arch: wasm32, os: wasi` regardless of the server's actual hardware or OS.

This is hitting us at @stackabletech since we use Krustlet-the-library to run native binaries.

**Comment by kflansburg** [comment]
I wonder if, since we don't know any better, we should remove the OS label from this list, which would allow the user to override it. 

https://github.com/krustlet/krustlet/blob/108579d28988d57a809ff74c27bd07ec9048328b/crates/kubelet/src/node/mod.rs#L458-L465
